migrations/20240910150600_create_search_notes_table.up.sql

```diff
@@ -0,0 +1,16 @@
+CREATE TABLE IF NOT EXISTS notes (
+    id                UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+
+    author_id         VARCHAR(255) NOT NULL,
+    note_id           UUID NOT NULL UNIQUE,
+
+    content           tsvector,
+    /* This field content both the target name and the public identifier */
+    target_name       tsvector,
+
+    updated_at        TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW()
+);
+
+--bun:split
+
+CREATE INDEX IF NOT EXISTS note_search ON notes USING GIN ( (notes.content || notes.target_name) );
```

Comment:
For the notes only, the index should also include `author_id`. You can modify this file directly, then `docker volume prune -f` when your infra is down to prevent local issues (this will reset your pg migrations).